src/fifelse.c

```diff
@@ -177,7 +179,11 @@ SEXP fcaseR(SEXP na, SEXP rho, SEXP args) {
       len0 = xlength(cons);
       len2 = len0;
       type0 = TYPEOF(outs);
-      value0 = outs;
+      v0class = PROTECT(getAttrib(outs,R_ClassSymbol)); nprotect++;
```

Comment:
couldn't we unprotect that some time sooner than at the very end of processing?

Reply:
I think it doesn’t matter considering the class and the levels are very small generally. In addition, we have to keep them to the end of the loop.